### PR TITLE
Improve query performance

### PR DESCRIPTION
**Pull request by batopa**
This PR slightly improves performance of query built in `BuildFilterBehavior::allowed_to_userFilter()`.

### CODE REVIEW

File: bedita-app/models/behaviors/build_filter.php [hunk 1]
```diff
@@ -1183,9 +1183,10 @@ protected function allowed_to_userFilter($s, $e, $userid) {
 
                 if (!empty($forbiddenPub) || !empty($forbiddenSection)) {
 
-                    $query = "SELECT {$s}Tree{$e}.{$s}id{$e}
+                    $query = "SELECT DISTINCT {$s}Tree{$e}.{$s}id{$e}
                              FROM {$s}trees{$e} AS {$s}Tree{$e}
-                             WHERE {$s}BEObject{$e}.{$s}id{$e} = {$s}Tree{$e}.{$s}id{$e}";
+                             INNER JOIN {$s}objects{$e} AS {$s}BEObject{$e} ON {$s}BEObject{$e}.{$s}id{$e} = {$s}Tree{$e}.{$s}id{$e}
+                             WHERE 1=1";
```
Comment (fquffio) [inline bedita-app/models/behaviors/build_filter.php:1189]:
```suggestion
                             INNER JOIN {$s}objects{$e} AS {$s}BEObject{$e} ON {$s}BEObject{$e}.{$s}id{$e} = {$s}Tree{$e}.{$s}id{$e}";
```

Should work without `WHERE` clause too…

File: bedita-app/models/behaviors/build_filter.php [hunk 2]
```diff
@@ -1208,9 +1209,9 @@ protected function allowed_to_userFilter($s, $e, $userid) {
                     $this->conditions[] = "(
                         (
                             NOT EXISTS (
-                                SELECT {$s}Tree{$e}.{$s}id{$e}
+                                SELECT DISTINCT {$s}Tree{$e}.{$s}id{$e}
                                 FROM {$s}trees{$e} AS {$s}Tree{$e}
-                                WHERE {$s}Tree{$e}.{$s}id{$e} = {$s}BEObject{$e}.{$s}id{$e}
+                                INNER JOIN {$s}objects{$e} AS {$s}BEObject{$e} ON {$s}BEObject{$e}.{$s}id{$e} = {$s}Tree{$e}.{$s}id{$e}
```
Comment (fquffio) [inline bedita-app/models/behaviors/build_filter.php:1214]: This changes the query meaning… it becomes a non-dependent subquery, which checks if _any_ object is on the tree. It is a condition that is basically never satisfied.

Comment (batopa) [reply bedita-app/models/behaviors/build_filter.php:1214]: I think you're right :disappointed:

**Comment by batopa** [comment]
No needed. I close it.

**Pull request closed without merging**